### PR TITLE
Declutter FD_WRITE and FD_READ

### PR DESCRIPTION
**Pull request by kinkie**
Avoid clashes with same-name constants defined by MS WIndows Socket API.

Also removed a few related ununsed macros.

### CODE REVIEW

**Review by yadij — DISMISSED** [review]

File: src/fd.h [hunk 1]
```diff
@@ -11,10 +11,16 @@
 #ifndef SQUID_FD_H_
 #define SQUID_FD_H_
 
+/// what kind of operation to count traffic for in fd_bytes()
```
Comment (rousskov) [inline src/fd.h:14]: We should also make it possible to reuse these values in other, similar contexts, without changing this description:

```suggestion
/// distinguishes reading/importing I/O operations from their writing/exporting counterparts
```

I do not insist on any specific wording.

File: compat/os/mswindows.h [hunk 2]
```diff
@@ -286,8 +286,6 @@ struct timezone {
 
 #undef FD_CLOSE
 #undef FD_OPEN
```
Comment (rousskov) [inline compat/os/mswindows.h:288]: If possible:

```suggestion
```

File: src/tests/stub_fd.cc [hunk 3]
```diff
@@ -18,7 +18,7 @@ fde *fde::Table = nullptr;
 int fdNFree(void) STUB_RETVAL(-1)
 void fd_open(int, unsigned int, const char *) STUB
 void fd_close(int) STUB
-void fd_bytes(int, int, unsigned int) STUB
+void fd_bytes(int, int, const FdOps) STUB
```
Comment (rousskov) [inline src/tests/stub_fd.cc:21]: There is no variable here for `const` to matter:

```suggestion
void fd_bytes(int, int, FdOps) STUB
```

**Review by rousskov — CHANGES_REQUESTED** [review]
Thank you for fixing this.

P.S. I will polish fd_bytes() implementation to match the new types.

File: src/fd.h [hunk 4]
```diff
@@ -11,10 +11,16 @@
...
+enum class FdOps {
+    READ,
+    WRITE
```
[the remark this answers is not in the document]
Comment (kinkie) [reply src/fd.h:17]: Done in 6cb1de634e (to be pushed shortly)

File: src/fd.h [hunk 5]
```diff
@@ -11,10 +11,16 @@
 #ifndef SQUID_FD_H_
 #define SQUID_FD_H_
 
+/// what kind of operation to count traffic for in fd_bytes()
+enum class FdOps {
```
[the remark this answers is not in the document]
Comment (kinkie) [reply src/fd.h:15]: Done in soon-to-be-pushed 4c522f9732

File: src/fd.h (hunk 1 above)
Comment (kinkie) [reply src/fd.h:14]: I don't have any preference, rewording as suggested

File: src/tests/stub_fd.cc (hunk 3 above)
Comment (kinkie) [reply src/tests/stub_fd.cc:21]: De-const-ed in to-be-pushed 73b392fa36

**Review by yadij — DISMISSED** [review]

**Review by rousskov — APPROVED** [review]
Thank you for this cleanup. Your changes allowed us to replace the runtime assertion with a compile-time check. I adjusted fd_bytes() accordingly, polished PR title/description, and approved this PR.

**Comment by rousskov** [comment]
Jenkins, test this please.

**Pull request closed without merging**